packages/core/src/app/shipping/Shipping.test.tsx

```diff
@@ -284,6 +286,221 @@ describe('Shipping step', () => {
+            checkout.use(CheckoutPreset.CheckoutWithMultiShipping);
+
+            const { container } = render(<CheckoutTest {...defaultProps} />);
+
+            await checkout.waitForShippingStep();
+
+            checkout.updateCheckout(
+                'post',
+                '/checkouts/xxxxxxxxxx-xxxx-xxax-xxxx-xxxxxx/consignments',
+                {
+                    ...checkoutWithBillingEmail,
+                    consignments: [
+                        {
+                            ...consignment,
+                            selectedShippingOption: undefined,
+                        },
+                    ],
+                },
+            );
+            checkout.updateCheckout(
+                'put',
+                '/checkouts/xxxxxxxxxx-xxxx-xxax-xxxx-xxxxxx/consignments/consignment-1',
+                {
+                    ...checkoutWithShipping,
+                },
+            );
+
+            checkout.updateCheckout(
+                'put',
+                '/checkouts/xxxxxxxxxx-xxxx-xxax-xxxx-xxxxxx/billing-address/billing-address-id*',
+                {
+                    ...checkoutWithShippingAndBilling,
+                },
+            );
+
+            expect(screen.getByTestId('address-select-button')).toBeInTheDocument();
+            await userEvent.click(screen.getByTestId('address-select-button'));
+            await userEvent.click(screen.getByTestId('add-new-address'));
+            await checkout.fillShippingAddress();
+
+            expect(checkoutService.updateShippingAddress).toHaveBeenCalled();
+            expect(
+                screen.getByRole('radio', { name: 'Pickup In Store $3.00' }),
+            ).toBeInTheDocument();
+            expect(screen.getByRole('radio', { name: 'Flat Rate $10.00' })).toBeInTheDocument();
+            // eslint-disable-next-line testing-library/no-container,testing-library/no-node-access
+            expect(container.getElementsByClassName('shippingOptions-skeleton').length).toBe(0);
+            // eslint-disable-next-line testing-library/no-container,testing-library/no-node-access
+            expect(container.getElementsByClassName('form-checklist-item--selected').length).toBe(
+                1,
+            );
+            // eslint-disable-next-line jest-dom/prefer-to-have-attribute
+            expect(
+                screen.getByLabelText('My billing address is the same as my shipping address.').hasAttribute('checked'),
+            ).toBeTruthy();
+
+            await userEvent.click(screen.getByRole('button', { name: 'Continue' }));
+
+            await checkout.waitForPaymentStep();
+
+            expect(checkoutService.updateBillingAddress).toHaveBeenCalled();
+            expect(screen.getByText(payments[0].config.displayName)).toBeInTheDocument();
+        });
+
+        it('selects the invalid customer address and completes the shipping step', async () => {
```

Comment:
Not sure if we need a better description here, as this feels like `user is able to complete shipping step with an invalid address` 🤔